Source/DiabloUI/diabloui.cpp

```diff
@@ -621,19 +621,27 @@ void UiDestroy()
 	UnloadUiGFX();
 }
 
-bool UiValidPlayerName(const char *name)
+bool UiValidPlayerName(string_view name)
 {
-	if (strlen(name) == 0)
+	if (name.empty())
 		return false;
 
-	if (strpbrk(name, ",<>%&\\\"?*#/:") != nullptr || strpbrk(name, " ") != nullptr)
+	// Currently only allow saving PLR_NAME_LEN bytes as a player name, so if the name is too long we'd have to truncate it.
+	// That said the input buffer is only 16 bytes long...
+	if (name.size() > PLR_NAME_LEN)
 		return false;
 
-	for (BYTE *letter = (BYTE *)name; *letter != '\0'; letter++)
-		if (*letter < 0x20 || (*letter > 0x7E && *letter < 0xC0))
+	if (name.find_first_of(",<>%&\\\"?*#/: ") != string_view::npos)
+		return false;
+
+	constexpr auto isAsciiControlCharacter = [](char character) -> bool {
+		return (character >= 0x00 && character < 0x20) || character == 0x7F;
+	};
+	for (auto letter = name.cbegin(); letter != name.cend(); letter++)
+		if (isAsciiControlCharacter(*letter) || !IsLeadUtf8CodeUnit(*letter))

```

Comment:
This also seems busted. The old logic didn't allow multibyte characters in multiplayer hero names since it was excluding any string containing a trailing character? There's no point allowing multibyte leads in that case...